backend/src/schema/resolvers/users.js

```diff
@@ -11,5 +11,27 @@ export default {
       params = await fileUpload(params, { file: 'avatarUpload', url: 'avatar' })
       return neo4jgraphql(object, params, context, resolveInfo, false)
     },
+    DeleteUser: async (object, params, context, resolveInfo) => {
+      const { resource } = params
+      const session = context.driver.session()
+
+      if (resource && resource.length) {
+        await Promise.all(
+          resource.map(async node => {
+            await session.run(
+              `
+            MATCH (resource:${node})<-[:WROTE]-(author:User {id: $userId})
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-07T22:17:19Z" title="Saturday, June 8th 2019, 12:17:19 am +02:00">Jun 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,5 +11,27 @@ export default {
       params = await fileUpload(params, { file: 'avatarUpload', url: 'avatar' })
       return neo4jgraphql(object, params, context, resolveInfo, false)
     },
+    DeleteUser: async (object, params, context, resolveInfo) => {
+      const { resource } = params
+      const session = context.driver.session()
+
+      if (resource && resource.length) {
+        await Promise.all(
+          resource.map(async node => {
+            await session.run(
+              `
+            MATCH (resource:${node})<-[:WROTE]-(author:User {id: $userId})
```


 A couple of thoughts here: First of all, I believe, you can do this in one statement. I would try sth. like:
```
MATCH (author:User {id: $userId}), (post:Post)<-[:WROTE]-(author), (comment:Comment)<[:WROTE]-(author)
DETACH DELETE author, post, comment
```
I don't know if it works like that but I guess you get the point.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-07T22:39:44Z" title="Saturday, June 8th 2019, 12:39:44 am +02:00">Jun 8, 2019</time>_

---


the point is they wouldn't have the choice of deleting just their account, or just their account and posts, just their account and comments, etc...

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-07T22:47:53Z" title="Saturday, June 8th 2019, 12:47:53 am +02:00">Jun 8, 2019</time>_

---


Oh, right, true! We cannot customize `neo4j-graphql-js`.